packages/staking/src/governance/api.ts

```diff
@@ -0,0 +1,62 @@
+import {AxiosInstance} from 'axios'
+import {axiosClient, GOVERNANCE_ENDPOINTS} from './config'
+
+export type GovernanceApi = {
+  getDRepById: (drepId: string) => Promise<{txId: string}>
+  isRegisteredDRep: (drepId: string) => Promise<boolean>
+}
+
+export const initApi = ({
+  networkId,
+  client = axiosClient,
+}: {
+  networkId: number
+  client?: Config['client']
+}) => {
+  return new Api({networkId, client})
+}
+
+class Api implements GovernanceApi {
+  constructor(private config: Config) {}
+
+  async getDRepById(drepId: string) {
+    const {networkId, client} = this.config
+    const backend =
+      networkId === 1
+        ? GOVERNANCE_ENDPOINTS.mainnet
+        : GOVERNANCE_ENDPOINTS.preprod
+
+    const response = await client.get<GetDRepByIdResponse>(
+      backend.getDRepById.replace('{{DREP_ID}}', drepId),
+    )
+
+    return {txId: response.data.registration.tx}
+  }
+
+  async isRegisteredDRep(drepId: string): Promise<boolean> {
+    try {
+      await this.getDRepById(drepId)
+      return true
+    } catch {
```

Comment:
is there a way to avoid try catch to business logic here?, cuz if we receive a 500 or a connection error it will "look" as DRep is not registered, it will need to deal with the error, I'm guessing if is not registered the api is returning 404, so only 404 should return false.